enhancements/windows-containers/csi-proxy.md

```diff
@@ -132,34 +132,111 @@ No action required.
+4.12 and 4.13. This specifically effects Azure, which loses in-tree storage functionality in 4.13, and vSphere which
+loses it in 4.14.
+
+What this means for the user, is that when they are upgrading a cluster from a version that uses in-tree to one that
+uses CSI, they must first upgrade WMCO to a version which runs csi-proxy on Windows Nodes. Only then should they
+upgrade the cluster.
+
+In order to help ensure a smooth transition WMCO v8 and v9 will block each individual Node upgrade if:
+1) A Node is being upgraded from in-tree storage to CSI driver storage
+2) A volume is attached to the Node
+
+The cluster administrator will need to apply the label `windowsmachineconfig.openshift.io/allow-upgrade=true` to each
+blocked Node, in order to unblock the upgrade. This way they can unblock an upgrade, and ensure that the workload is
+able to go back into running, using the migrated storage volume. If the workload does not go back into running, the
+cluster admin must debug any issues with the CSI drivers, and correct them, before allowing upgrades on any other
+Nodes. This is meant to minimize any workload interruption on upgrade.
+
+##### Possible upgrade path scenarios
+
+###### v7 -> v8
+
+Scenario A:
+
+v7 with csi-proxy backport -> v8 with csi-proxy backport
+
+csi-proxy will be running on each Node. WMCO v8 will be installed on the cluster, and the individual Node upgrades will
+be blocked for Azure, which must have the storage volumes migrated. The user will be able to deploy the CSI drivers
+and confirm they go into running. The user will be able to follow the ideal upgrade procedure.
+
+Scenario B:
+
+v7 without csi-proxy backport -> v8 with csi-proxy backport
+
+csi-proxy will not be running on each Node. WMCO 8 will be installed on the cluster, and the individual Node upgrades
+will be blocked for Azure, which must have the storage volumes migrated. The user will be able to deploy the CSI
+drivers, but will not be able to confirm the CSI Windows daemonset pods go into running until a Node has been upgraded.
+The user should upgrade a Node, confirm the csi-driver Windows daemonset pod for the node goes into running, and then
+confirm that the workload goes into running.
+
+Scenario C:
+
+v7 with csi-proxy backport -> v8 without csi-proxy backport
+
+csi-proxy will initally be running on each Node. WMCO 8 will be installed on the cluster, and the individual Node
+upgrades will not be blocked, and csi-proxy will stop running on each Node.
+For vSphere clusters, there is no impact as in-tree storage will still be used.
+For Azure and all other platforms, neither in-tree volumes nor CSI volumes will be usable. Storage workloads will be
+disrupted until WMCO and all Windows Nodes are upgraded to a version of v8 with the csi-proxy backport.
+
+Scenario C needs to be avoided. For this reason WMCO v8 with csi-proxy backport should be released before WMCO v7
```

Comment:
Does this imply that the release ordering is v9, v8 and v7? I thought we decided that it should be the other way around?

Reply:
Yes I have come to the conclusion that in general while releasing  v7 first and then v8 would be smoother in the general case, in order to avoid Scenario C, which can cause major issues we should release v8 first. I did not find anything that would cause us to care about the v9 release timing, please let me know if you think I'm missing something there.

Reply:
What if a customer is on v8 with CSI proxy and then upgrades to v9 where CSI proxy is not available?

Reply:
v9 will only be released with csi-proxy, we haven't released a version without it, unlike v7 and v8.

Reply:
Good point. What about upgrading from v8 version without csi-proxy to v9 which has csi-proxy?

Reply:
That is Scenario B (Same as Scenario E)

Reply:
OK, what is tripping me up is which version is going to block the upgrades. @sebsoto I suggest explicitly calling that out in the scenarios.

Reply:
Reworded things to be explicit about that